services/eth_gateway.go

```diff
@@ -212,17 +210,25 @@ func getCurrentBlock() (*types.Block, error) {
 // WaitForTransfer is blocking call that will observe on brokerAddr on transfer on ETH.
 // If it is completed return true, otherwise, return false (or time-out)
```

Comment:
Instead of returning booleans, would it be an option in some of these places to just do 'return err' so it's obvious to someone watching the broker what the error was?  Something else that would be really helpful is to send all these errors to Sentry so we can track them.  You can to that like this:  
		if err != nil {
			raven.CaptureError(err, nil)
                        return err
		}

Reply:
I think in some places, it makes sense to return boolean rather than an error. Since caller does not really care about the error and it won't be able to do much. 

But in a case by case, any method you think we should return err instead, please let me know.

For example, sendETH return a string, instead, it should return string, error. I am planning to fix it in the next PR.

Reply:
I agree with Rebel. Returning an err is more clear to the caller that the function can fail. A success boolean isn't clear from just reading the function signature. Checking for err seems to be a Golang idiom too.

Reply:
I'm going to make a task to add Sentry logging throughout the brokernode.  There's quite a few files where we aren't going it, and we probably should be doing it everywhere.  It'll help us a lot in the coming week.